### PR TITLE
Add a work-around for memsan

### LINKED ISSUE #11735: Use of uninitialized value in `fd_getcrtime_at`
#11730 

```
==20268==WARNING: MemorySanitizer: use-of-uninitialized-value
    #0 0x7fadff72275c in fd_getcrtime_at /build/build/../src/basic/xattr-util.c:165:41
    #1 0x7fadff72426f in fd_getcrtime /build/build/../src/basic/xattr-util.c:197:16
    #2 0x7fadff30c30a in copy_times /build/build/../src/basic/copy.c:869:13
    #3 0x7fadff30b5db in copy_file_fd_full /build/build/../src/basic/copy.c:746:16
    #4 0x7fadff30f5ea in copy_file_full /build/build/../src/basic/copy.c:776:13
    #5 0x4a5d74 in copy_file /build/build/../src/basic/copy.h:29:16
    #6 0x497640 in test_copy_file /build/build/../src/test/test-copy.c:41:9
    #7 0x4964d1 in main /build/build/../src/test/test-copy.c:261:9
    #8 0x7fadfe79d09a in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2409a)
    #9 0x41f469 in _start (/build/build/test-copy+0x41f469)

  Uninitialized value was created by an allocation of 'sx' in the stack frame of function 'fd_getcrtime_at'
    #0 0x7fadff721ae0 in fd_getcrtime_at /build/build/../src/basic/xattr-util.c:142

SUMMARY: MemorySanitizer: use-of-uninitialized-value /build/build/../src/basic/xattr-util.c:165:41 in fd_getcrtime_at
Exiting
```

https://github.com/systemd/systemd/blob/master/src/basic/xattr-util.c#L142-L171

(I don't see anything wrong in this particular section, but I'm probably missing something pretty obvious).

### PR DESCRIPTION
**Pull request by keszybz**
Fixes #11735.

### CODE REVIEW

File: src/basic/xattr-util.c [hunk 1]
```diff
@@ -140,7 +140,12 @@ static int parse_crtime(le64_t le, usec_t *usec) {
 }
 
 int fd_getcrtime_at(int dirfd, const char *name, usec_t *ret, int flags) {
-        struct_statx sx;
+        struct_statx sx
+#if HAS_FEATURE_MEMORY_SANITIZER
+                = {} /* This is just for the memory sanitizer. Remove after it learns
+                      * about statx. */
```
Comment (poettering) [inline src/basic/xattr-util.c]: maye add a `#warn` here?

Comment (keszybz) [reply src/basic/xattr-util.c]: What would I warn about? Running under msan is not an error.

Comment (poettering) [reply src/basic/xattr-util.c]: just that as a reminder that this is still in there, and should be removed one day. i.e.

```
#warning "Initializing struct statx explicitly to zero, to work around msan limitation. Please remove as soon as msan has been updated to not need this any longer"
```

**Comment by keszybz** [comment]
On Thu, Feb 21, 2019 at 02:15:42AM -0800, Lennart Poettering wrote:
> poettering commented on this pull request.
> 
> 
> 
> > @@ -140,7 +140,12 @@ static int parse_crtime(le64_t le, usec_t *usec) {
>  }
>  
>  int fd_getcrtime_at(int dirfd, const char *name, usec_t *ret, int flags) {
> -        struct_statx sx;
> +        struct_statx sx
> +#if HAS_FEATURE_MEMORY_SANITIZER
> +                = {} /* This is just for the memory sanitizer. Remove after it learns
> +                      * about statx. */
> 
> just that as a reminder that this is still in there, and should be removed one day. i.e.
> 
> ```
> #warning "Initializing struct statx explicitly zero, to work around msan limitation. Please remove as soon as msan has been updated to not need this any longer"

Ah, a gentle hint ;)
Updated.

**Comment by poettering** [comment]
thanks! lgtm

**Pull request merged**